setup.sh

```diff
@@ -26,343 +36,288 @@ WARN="${YELLOW}WARNING:$RESET"
-	fi
-}
-
-is_network_configured() {
-	ip addr show $TINKERBELL_PROVISIONER_INTERFACE | grep $TINKERBELL_HOST_IP >>/dev/null && ip addr show $TINKERBELL_PROVISIONER_INTERFACE | grep $TINKERBELL_NGINX_IP >>/dev/null
-}
-
-write_iface_config() {
-	iface_config="$(
-		cat <<EOF | sed 's/^\s\{4\}//g' | sed ':a;N;$!ba;s/\n/\\n/g'
-    auto $TINKERBELL_NETWORK_INTERFACE:0
-    iface $TINKERBELL_NETWORK_INTERFACE:0 inet static
-        address $TINKERBELL_HOST_IP
-        netmask $TINKERBELL_NETMASK
-        broadcast $TINKERBELL_BROADCAST_IP
-        pre-up sleep 4
-
-    auto $TINKERBELL_NETWORK_INTERFACE:1
-    iface $TINKERBELL_NETWORK_INTERFACE:1 inet static
-        address $TINKERBELL_NGINX_IP
-        netmask $TINKERBELL_NETMASK
-        broadcast $TINKERBELL_BROADCAST_IP
-        pre-up sleep 4
 EOF
-	)"
-	sed -i "/^auto $TINKERBELL_NETWORK_INTERFACE/,/^\$/c $iface_config" /etc/network/interfaces
-}
-
-setup_networking() {
-	if ! is_network_configured; then
-		cidr=$(echo $TINKERBELL_NETWORK | grep -Eo "\/[[:digit:]]+" | grep -v "^$" | tr -d "/")
-		case "$1" in
-		ubuntu)
-			if [ ! -f /etc/network/interfaces ]; then
-				echo "$ERR file /etc/network/interfaces not found"
-				exit 1
-			fi
+)
 
-			if grep -q $TINKERBELL_HOST_IP /etc/network/interfaces; then
-				echo "$INFO tinkerbell network interface is already configured"
-			else
-				# plumb IP and restart to tinkerbell network interface
-				if grep -q $TINKERBELL_NETWORK_INTERFACE /etc/network/interfaces; then
-					echo "" >>/etc/network/interfaces
-					write_iface_config
-				else
-					echo -e "\nauto $TINKERBELL_NETWORK_INTERFACE\n" >>/etc/network/interfaces
-					write_iface_config
-				fi
-				ip link set $TINKERBELL_NETWORK_INTERFACE nomaster
-				ifdown "$TINKERBELL_NETWORK_INTERFACE:0"
-				ifdown "$TINKERBELL_NETWORK_INTERFACE:1"
-				ifup "$TINKERBELL_NETWORK_INTERFACE:0"
-				ifup "$TINKERBELL_NETWORK_INTERFACE:1"
-			fi
-			;;
-		centos)
-			if [ -f /etc/sysconfig/network-scripts/ifcfg-$TINKERBELL_NETWORK_INTERFACE ]; then
-				sed -i '/^ONBOOT.*no$/s/no/yes/; /^BOOTPROTO.*none$/s/none/static/; /^MASTER/d; /^SLAVE/d' /etc/sysconfig/network-scripts/ifcfg-$TINKERBELL_NETWORK_INTERFACE
-			else
-				touch /etc/sysconfig/network-scripts/ifcfg-$TINKERBELL_NETWORK_INTERFACE
-				HWADDRESS=$(ip addr show $TINKERBELL_NETWORK_INTERFACE | grep ether | awk -F 'ether' '{print $2}' | cut -d" " -f2)
-				cat <<EOF >>/etc/sysconfig/network-scripts/ifcfg-$TINKERBELL_NETWORK_INTERFACE
+setup_networking_centos() (
+	HWADDRESS=$(ip addr show "$TINKERBELL_NETWORK_INTERFACE" | grep ether | awk -F 'ether' '{print $2}' | cut -d" " -f2)
+	content=$(
+		cat <<EOF
 DEVICE=$TINKERBELL_NETWORK_INTERFACE
 ONBOOT=yes
 HWADDR=$HWADDRESS
 BOOTPROTO=static
-EOF
-			fi
 
-			cat <<EOF >>/etc/sysconfig/network-scripts/ifcfg-$TINKERBELL_NETWORK_INTERFACE
 IPADDR0=$TINKERBELL_HOST_IP
 NETMASK0=$TINKERBELL_NETMASK
 IPADDR1=$TINKERBELL_NGINX_IP
 NETMASK1=$TINKERBELL_NETMASK
```

Comment:
Can use just PREFIX vs NETMASK. If we drop NETMASK/Netmask in deb and rh based configs we don't need to ask the user to calculate.

Reply:
Great idea, thanks, sorted.